core/src/main/java/com/xmartlabs/suspendedlocation/core/FlowLocationCallback.kt

```diff
@@ -0,0 +1,37 @@
+package com.xmartlabs.suspendedlocation.core
+
+import android.location.Location
+import com.google.android.gms.location.LocationAvailability
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationResult
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.ProducerScope
+import kotlinx.coroutines.runBlocking
+import java.io.IOException
+import kotlin.coroutines.CoroutineContext
+
+@ExperimentalCoroutinesApi
+internal class FlowLocationCallback(
+    val producerScope: ProducerScope<Location>,
+    val runnerContext: CoroutineContext
+) : LocationCallback() {
+  @Synchronized
+  override fun onLocationResult(locationResult: LocationResult) {
+    runBlocking(runnerContext) {
+      producerScope.send(locationResult.lastLocation)
+    }
+  }
+
+  override fun onLocationAvailability(locationAvailability: LocationAvailability) {
+    super.onLocationAvailability(locationAvailability)
+    runBlocking(runnerContext) {
+      if (!locationAvailability.isLocationAvailable) {
+        producerScope.cancel(
```

Comment:
what happens if you lost and then if you have the connection again, your flow will not continue right? It should continue

Reply:
why? i think that as it is right now is correct since it binds to the behavior of the google places API, since we don't know what cause why the location is not available.